internal/rpcserver/rpcserver.go

```diff
@@ -3686,10 +3686,14 @@ func handleGetWorkRequest(s *Server) (interface{}, error) {
 	if template == nil {
 		var err error
 		template, err = bt.CurrentTemplate()
-		if err != nil || template == nil {
+		if err != nil {
 			context := "Unable to retrieve work due to invalid template"
 			return nil, rpcInternalError(err.Error(), context)
 		}
+		if template == nil {
```

Comment:
This is already guarded by `template == nil`, so no need to check again.

Reply:
See line 3688, it updates `template` which typically will make it non-nil, but the specific case in question here is when it doesn't.

Reply:
Ah, right.  Disregard that comment.